layers/eight_mile/tf/layers.py

```diff
@@ -848,6 +1094,8 @@ def output_fn(self, rnnout, state):
     def call(self, inputs):
         inputs, lengths = tensor_and_lengths(inputs)
         mask = tf.sequence_mask(lengths)
+        max_length = tf.reduce_max(lengths)
+        inputs = inputs[:, :max_length, :]
```

Comment:
This seems overly prescriptive to the `[B, T, H]` shape. Does it make sense for a function that takes a dim and only culls on the dim specified? Right now this would fail at cutting a tensor that was like `[B, T, H, H2]` or something

Reply:
would that make sense for an RNN input though?

Reply:
Not right now but one of the pain points in dynet was that was hard to handle things that had extra dimension like the transformer attention so figured we would want to move away from things being so locked into the `[B, T, H]` mindset

Reply:
we can engineer it more when there are less redundant things